File: mss-product/entities/product.js
class Product {
  constructor({ code, name, price, type, description, urlPhoto }) {
    this.code = code;
    this.name = name;
    this.price = price;
    this.type = type;
    this.description = description;
    this.urlPhoto = urlPhoto;
  }

  toJSON() {
    return {
      code: this.code,
      name: this.name,
      price: this.price,
      type: this.type,
      description: this.description,
      urlPhoto: this.urlPhoto,
    };
  }
}

module.exports = Product;

File: frontEnd/front_loja_online_flutter/lib/src/entities/product.dart
class Product {
  final int code;
  final String name;
  final String urlPhoto;
  final double price;

  Product({
    required this.code,
    required this.name,
    required this.urlPhoto,
    required this.price,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      code: int.parse(json['code'] as String),
      name: json['name'] as String,
      urlPhoto: json['urlPhoto'] as String,
      price: json['price'] as double,
    );
  }

  static List<Product> listFromJson(List<dynamic> json) {
    return json
        .map((item) => Product.fromJson(item as Map<String, dynamic>))
        .toList();
  }
}
